controllers/profiles.js

```diff
@@ -1,26 +1,33 @@
-const express = require('express');
+const express = require("express");
 const router = express.Router();
-const User = require('../models/user');
-const verifyToken = require('../middleware/verify-token');
+const User = require("../models/user");
+const { Post: Pinsta } = require("../models/pinsta");
+const verifyToken = require("../middleware/verify-token");
 
-router.get('/:userId', verifyToken, async (req, res) => {
-    try {
-        if (req.user._id !== req.params.userId){
-            return res.status(401).json({ error: "Unauthorized"})
-        }
-        const user = await User.findById(req.user._id);
-        if (!user) {
-            res.status(404)
-            throw new Error('Profile not found.');
-        }
-        res.json({ user });
-    } catch (error) {
-        if (res.statusCode === 404) {
-            res.status(404).json({ error: error.message });
-        } else {
-            res.status(500).json({ error: error.message });
-        }
+// GET - Show Profile with User's Posts
+router.get("/:userId", verifyToken, async (req, res) => {
+  try {
+    if (req.user._id.toString() !== req.params.userId) {
+      return res.status(401).json({ error: "Unauthorized" });
     }
+
+    // Fetch the user profile
+    const user = await User.findById(req.user._id);
+    if (!user) {
+      res.status(404);
+      throw new Error("Profile not found.");
+    }
+    // Fetch user’s posts
+    const userPosts = await Pinsta.find({ userId: req.user._id });
```

Comment:
This should be changed to `const userPosts = await Pinsta.find({ author_id: req.user._id });`, because we are using `author_id` in Pinsta Model, instead of `userId`.